src/main/java/de/tum/in/www1/artemis/service/connectors/localci/buildagent/BuildJobContainerService.java

```diff
@@ -189,6 +189,11 @@ public void stopContainer(String containerName) {
         // for some other reason.
         // The container's main process is waiting for this file to appear and then stops the main process, thus stopping and removing the container.
         executeDockerCommandWithoutAwaitingResponse(containerId, "touch", LOCALCI_WORKING_DIRECTORY + "/stop_container.txt");
+
+        // If the container is still running, force kill it.
+        containers = dockerClient.listContainersCmd().withShowAll(true).exec();
+        containerOptional = containers.stream().filter(container -> container.getNames()[0].equals("/" + containerName)).findFirst();
+        containerOptional.ifPresent(container -> dockerClient.removeContainerCmd(container.getId()).withForce(true).exec());
     }
```

Comment:
Does it make sense to directly check this here? Especially since the method above is not waiting for the command to finish. Maybe we can schedule this to execute a few seconds after the normal stop command?

Reply:
I talked to Stephan and refactored the logic. It looks different now